usr.sbin/bhyve/rtc_pl031.c

```diff
@@ -275,9 +275,9 @@ rtc_pl031_init(rtc_pl031_intr_func_t intr_assert,
 	sc->intr_assert = intr_assert;
 	sc->intr_deassert = intr_deassert;
 
-	rtc_pl031_update_timer(sc);
 	sc->mevp = mevent_add(rtc_pl031_next_timer_msecs(sc), EVF_TIMER,
 	    rtc_pl031_callback, sc);
+	rtc_pl031_update_timer(sc);
```

Comment:
I don’t think this is even needed? It calls mevent_add with rtc_pl031_next_timer_Marc’s already, and update_timer just uses the exact same value (minus however much time has passed).

Reply:
I think what happened is I used to have update_timer do the mevent_add itself if mevp was NULL, but then hoisted it out to here whilst forgetting to delete the call.

Reply:
Indeed, you're right.